src/metabase/models/secret.clj

```diff
@@ -100,6 +101,24 @@
           (.write out v)))
       tmp-file)))
 
+(def
+  ^java.io.File
+  ^{:arglists '([{:keys [connection-property-name id value] :as secret} driver?])}
```

Comment:
maybe too late to fix this but the `:tag` metadata should be on the argslist, not on `#'value->file!` itself. Otherwise you're basically saying that `value->file!` itself is an instance of `java.io.File`, rather than the result of invoking it as a function.

The Clojure compiler is smart enough to figure out what you mean if you use this as a function anyway but it can potentially not error with a reflection warning if you did something like 

```clj
(.someMethodThatWouldCauseAReflectionWarning value->file!)
```

when you mean to do

```clj
(.someMethodThatWouldCauseAReflectionWarning (value->file! x))
```

instead